### DISCUSSION TITLE
Caching based on response

### DISCUSSION THREAD

**Discussion by bairrada97** [vercel/next.js#54833] [App Router]
I see that next js team put a lot of effort on making caching more robust and with the new extensive docs about caching https://nextjs.org/docs/app/building-your-application/caching  which I think its pretty good. But I dont see any example or if its even possible to cache based on the response.

I want to get rid of my express server but at the moment I can't seem to get rid of it purely because of the cache.

my use case:
I have a list of events and if I go to event detail I fetch data for that specific event.

- If the event is live I want to cache it for 15 seconds 
- if the event already finished I want to cache for 1 week
- if the event is in the future I want to cache until the event starts. 

All of this to limit fetching data from third party api.
 
 In the docs all I see is caching with a hardcoded value `fetch('https://...', { next: { revalidate: 3600 } })` but never based on the response.
 
 Currently Im using an express server using `node-cache` just for this purpose but I would like to use next js for this aswell

**Comment by jbergstroem** [comment]
I also want this! https://github.com/vercel/next.js/discussions/54684
